File: app/src/main/java/com/hu/p7zip/ZipUtils.java
package com.hu.p7zip;

/**
 * Created by LJIAN on 2017/5/11.
 */
public class ZipUtils {

    /**
     * Execute a command
     * @param command command string
     * @return return code
     */
    public static native int executeCommand(String command);

    /**
     * load native library
     */
    static {
        System.loadLibrary("p7zip");
    }
}

File: app/src/main/java/com/hu/p7zip/ZipUtils.kt
package com.hu.p7zip

/**
 * Created by LJIAN on 2017/5/11.
 */
object ZipUtils {

    /**
     * Execute a command
     * @param command command string
     * @return return code
     */
    external fun executeCommand(command: String): Int

    /**
     * load native library
     */
    init {
        System.loadLibrary("p7zip")
    }
}
